solutions/swb-reference/end-to-end-tests/cypress/support/commands.ts

```diff
@@ -0,0 +1,90 @@
+/// <reference types="cypress" />
+
+// ***********************************************
+// This example commands.ts shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add('login', (email, password) => { ... })
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+//
+// declare global {
+//   namespace Cypress {
+//     interface Chainable {
+//       login(email: string, password: string): Chainable<void>
+//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
+//     }
+//   }
+// }
+import {
+  ADMIN_PASSWORD_PROPERTY,
+  ADMIN_USER_PROPERTY,
+  BASE_URL_PROPERTY,
+  COGNITO_DOMAIN_NAME_PROPERTY
+} from './constants';
+
+Cypress.on(
+  'uncaught:exception',
+  (err) => !err.message.includes('ResizeObserver loop completed with undelivered notifications') //known issue with cypress
+);
+
+Cypress.Commands.add('login', (role: string) => {
+  const login = getLoginInfo(role);
+
+  Cypress.config('baseUrl', Cypress.env(BASE_URL_PROPERTY));
+  cy.visit('/');
+  cy.get('#header button').contains('Researcher User');
+  cy.get('[data-testid="login"]').should('be.visible');
+  cy.get('[data-testid="login"]').click();
+  cy.origin(
+    Cypress.env(COGNITO_DOMAIN_NAME_PROPERTY),
+    { args: [login.user, login.password] },
+    ([user, password]) => {
+      cy.get('input[name=username]:visible', { timeout: 10000 }).type(user); //wait up to 10 seconds to have the username field displayed
+      cy.get('[name=password]:visible').type(password);
+      cy.get('[name=signInSubmitButton]:visible').click();
+    }
+  );
+
+  cy.wait(5000); //redirection for environments may take time to load
```

Comment:
Instead of setting a wait time, we should wait for a specific attribute to appear. For example, you can do a `cy.get` for the environments page header. If it doesn't appear after 5 seconds then the test fails. This way the tests can potentially wait for less than 5 seconds.

An example is shown here
https://github.com/awslabs/service-workbench-on-aws/blob/4a7700033891a76a68826546d6ddda5cdb1bcf8a/main/end-to-end-tests/cypress/support/commands.js#L102

Reply:
wait commands removed